src/Core/Form/IdentifiableObject/Builder/FormBuilder.php

```diff
@@ -89,7 +98,17 @@ public function __construct(
      */
     public function getForm(array $data = [], array $options = [])
     {
-        if (is_array($defaultData = $this->dataProvider->getDefaultData())) {
+        $defaultData = $this->dataProvider->getDefaultData();
+
+        $this->hookDispatcher->dispatchWithParameters(
+            'action' . $this->registry->getType($this->formType)->getBlockPrefix() . 'FormDataProviderDefaultData',
+            [
+                'defaultData' => &$defaultData,
+                'options' => $options,
```

Comment:
```suggestion
                'options' => &$options,
```

Options should be editable too

Reply:
Fixed.